xcdat/temporal.py

```diff
@@ -968,6 +968,7 @@ def _get_weights(self) -> xr.DataArray:
         # only select the general DType and not details such as the byte order
         # or time unit (with rare exceptions see release notes). To avoid this
         # warning please use the scalar types `np.float64`, or string notation.`
+        time_lengths.load()
```

Comment:
We should do `.load()` only if it is a Dask Array (same logic as in spatial averaging).

---

Add this import at the top of the file: `from dask.array.core import Array`

```suggestion
        if type(d_bounds.data) == Array:
        	time_lengths.load()
```

Reply:
I think we'll need one additional test for this conditional statement, similar to https://github.com/xCDAT/xcdat/blob/8544a43ce7fefff718327c2cc9a2e95aedd0e918/tests/test_spatial.py#L170-L193.

Reply:
@tomvothecoder - is this existing test sufficient?